src/main/java/org/candlepin/insights/exception/ErrorCode.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.insights.exception;
+
+/**
+ * Represents the various error codes that can be encountered by callers
+ * of the API.
+ */
+public enum ErrorCode {
+
+    /**
+     * An unhandled exception has occurred. This typically implies that the
+     * exception was unexpected and likely due to a bug or coding error.
+     */
+    UNHANDLED_EXCEPTION_ERROR(100, "An unhandled exception occurred"),
+
+    /**
+     * An exception was thrown by RestEasy while processing a request to the
+     * rhsm-conduit API. This typically means that an HTTP client error has
+     * occurred (HTTP 4XX).
+     */
+    REQUEST_PROCESSING_ERROR(101, "An error occurred while processing a request."),
+
+    /**
+     * An unexpected exception was thrown by the inventory service client.
+     */
+    INVENTORY_SERVICE_ERROR(200, "Inventory Service Error"),
+
+    /**
+     * The inventory service is unavailable. This typically means that the
+     * inventory service is either down, or the configured service URL is
+     * not correct, and a connection can not be made.
+     */
+    INVENTORY_SERVICE_UNAVAILABLE(201, "The inventory service is unavailable"),
+
+    /**
+     * An exception was thrown by the inventory service when a request was made.
+     * This typically means that an HTTP client error has occurred (HTTP 4XX)
+     * when rhsm-conduit made the request.
+     */
+    INVENTORY_SERVICE_REQUEST_ERROR(202, "Inventory API Error"),
+
+    /**
+     * An unexpected exception was thrown by the Pinhead service client.
+     */
+    PINHEAD_SERVICE_ERROR(300, "Pinhead Service Error");
```

Comment:
Is `300` right?  That's kind of a weird status code

Reply:
These are Application codes NOT http codes.

Will likely move the prefix into the ErrorCode class.

Reply:
And also go with 4 digit codes like RHSMCONDUIT3000, RHSMCONDUIT3001